tox.ini

```diff
@@ -2,25 +2,21 @@
 envlist = py{27,34,35,36,py}-{brief,prettyquick,full,custom,benchmark}
 toxworkdir={env:TOX_WORK_DIR:.tox}
 
-passenv=
-  HOME
-  LC_ALL
-  COVERAGE_FILE
-
-setenv=
-  PYTHONWARNINGS={env:PYTHONWARNINGS:error::DeprecationWarning,error::FutureWarning}
-
 [testenv]
 deps =
     -rrequirements/test.txt
     benchmark: pytest-benchmark==3.0.0
 whitelist_externals=
     bash
-setenv=
-    brief: HYPOTHESIS_PROFILE=speedy
 passenv=
     HOME
+    LC_ALL
+    COVERAGE_FILE
     TOXENV
+setenv=
+    ; Need to ignore warnings from pandas._version: https://github.com/pandas-dev/pandas/issues/19944
+    PYTHONWARNINGS={env:PYTHONWARNINGS:error::FutureWarning,default::FutureWarning:pandas._version}
```

Comment:
I think we actually want to turn all warnings into errors, except for FutureWarning from Pandas or any DeprecationWarning:

    PYTHONWARNINGS={env:PYTHONWARNINGS:error,default::FutureWarning:pandas._version,default::DeprecationWarning}

Reply:
Hmm, with that `make check-py36` fails with:

<details>
make check-py36
rm -f /Users/davidchudzicki/.cache/hypothesis-build-runtimes/tools/tox
ln -sf /Users/davidchudzicki/.cache/hypothesis-build-runtimes/virtualenvs/tools-f144d01447/bin/tox /Users/davidchudzicki/.cache/hypothesis-build-runtimes/tools/tox
touch /Users/davidchudzicki/.cache/hypothesis-build-runtimes/virtualenvs/tools-f144d01447/bin/tox /Users/davidchudzicki/.cache/hypothesis-build-runtimes/tools/tox
/Users/davidchudzicki/.cache/hypothesis-build-runtimes/tools/tox --recreate -e py36-full
GLOB sdist-make: /Users/davidchudzicki/hypothesis-python/setup.py
py36-full recreate: /Users/davidchudzicki/.cache/hypothesis-build-runtimes/.tox/py36-full
py36-full installdeps: -rrequirements/test.txt
py36-full inst: /Users/davidchudzicki/.cache/hypothesis-build-runtimes/.tox/dist/hypothesis-3.49.1.zip
py36-full installed: /Users/davidchudzicki/.cache/hypothesis-build-runtimes/.tox/py36-full/lib/python3.6/distutils/__init__.py:4: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses,  import imp,apipkg==1.4,attrs==17.4.0,coverage==4.5.1,execnet==1.5.0,flaky==3.4.0,hypothesis==3.49.1,mock==2.0.0,pbr==3.1.1,pluggy==0.6.0,py==1.5.2,pytest==3.4.2,pytest-forked==0.2,pytest-xdist==1.22.2,six==1.11.0
py36-full runtests: PYTHONHASHSEED='4066271767'
py36-full runtests: commands[0] | bash scripts/basic-test.sh
++ uname -s
+ [[ Darwin == \D\a\r\w\i\n ]]
+ DARWIN=true
+ python -c '
import os
for k, v in sorted(dict(os.environ).items()):
    print("%s=%s" % (k, v))
'
COVERAGE_FILE=/Users/davidchudzicki/.cache/hypothesis-build-runtimes/.coverage
HOME=/Users/davidchudzicki
LANG=en_US.UTF-8
LC_ALL=en_US.UTF-8
PATH=/Users/davidchudzicki/.cache/hypothesis-build-runtimes/.tox/py36-full/bin:/Users/davidchudzicki/.cache/hypothesis-build-runtimes/snakepit:/Users/davidchudzicki/.cache/hypothesis-build-runtimes/tools:/Users/davidchudzicki/.virtualenvs/hypothesis/bin:/Library/Frameworks/Python.framework/Versions/3.6/bin:/Users/david/miniconda2/bin:/Users/david/miniconda2/bin:/Users/david/miniconda2/bin:/Users/david/miniconda2/bin:/Users/david/miniconda2/bin:/Users/davidchudzicki/.gem/ruby/2.0.0/bin:/Users/david/github/makefile2graph:/Users/david/miniconda/bin:/Users/david/miniconda3/bin/:/Users/davidchudzicki/bin:~/Dropbox/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/X11/bin:/Library/TeX/texbin:~/Kaggle.RUtilities/bin:/usr/texbin/:/Applications/Julia-0.3.9.app/Contents/Resources/julia/bin/:/Library/Frameworks/R.framework/Versions/3.2/Resources/bin:/usr/local/lib/node_modules/jlbox/node_modules/gulp/bin:/usr/local/Cellar/imagemagick/6.8.7-7/bin
PWD=/Users/davidchudzicki/hypothesis-python
PYTHONHASHSEED=4066271767
PYTHONWARNINGS=error,default::FutureWarning:pandas._version,default::DeprecationWarning
SHLVL=1
TMPDIR=/var/folders/8l/psdmt62d0wzfhdr6lc4fvyb00000gn/T/
VIRTUAL_ENV=/Users/davidchudzicki/.cache/hypothesis-build-runtimes/.tox/py36-full
_=/Users/davidchudzicki/.cache/hypothesis-build-runtimes/.tox/py36-full/bin/python
__CF_USER_TEXT_ENCODING=0x1F5:0x0:0x0
+ pip install .
/Users/davidchudzicki/.cache/hypothesis-build-runtimes/.tox/py36-full/lib/python3.6/distutils/__init__.py:4: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  import imp
Processing /Users/davidchudzicki/hypothesis-python
Exception ignored in: <_io.FileIO name=3 mode='rb' closefd=True>
ResourceWarning: unclosed file <_io.BufferedReader name=3>
  Requirement already satisfied (use --upgrade to upgrade): hypothesis==3.49.1 from file:///Users/davidchudzicki/hypothesis-python in /Users/davidchudzicki/.cache/hypothesis-build-runtimes/.tox/py36-full/lib/python3.6/site-packages
Requirement already satisfied: attrs>=16.0.0 in /Users/davidchudzicki/.cache/hypothesis-build-runtimes/.tox/py36-full/lib/python3.6/site-packages (from hypothesis==3.49.1)
Requirement already satisfied: coverage in /Users/davidchudzicki/.cache/hypothesis-build-runtimes/.tox/py36-full/lib/python3.6/site-packages (from hypothesis==3.49.1)
Building wheels for collected packages: hypothesis
  Running setup.py bdist_wheel for hypothesis ... done
Exception ignored in: <_io.FileIO name=3 mode='rb' closefd=True>
ResourceWarning: unclosed file <_io.BufferedReader name=3>
  Stored in directory: /Users/davidchudzicki/Library/Caches/pip/wheels/92/ef/b8/020db7f11d54e14f3b192c359956ca4995b870aa471a575d1e
Successfully built hypothesis
+ PYTEST='python -m pytest -n2'
+ python -m pytest -n2 --runpytest=subprocess tests/pytest
/Users/davidchudzicki/.cache/hypothesis-build-runtimes/.tox/py36-full/lib/python3.6/site-packages/_pytest/assertion/rewrite.py:6: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  import imp
/Users/davidchudzicki/.cache/hypothesis-build-runtimes/.tox/py36-full/lib/python3.6/site-packages/_pytest/fixtures.py:847: DeprecationWarning: The `convert` argument is deprecated in favor of `converter`.  It will be removed after 2019/01.
  params = attr.ib(convert=attr.converters.optional(tuple))
/Users/davidchudzicki/.cache/hypothesis-build-runtimes/.tox/py36-full/lib/python3.6/site-packages/_pytest/fixtures.py:849: DeprecationWarning: The `convert` argument is deprecated in favor of `converter`.  It will be removed after 2019/01.
  ids = attr.ib(default=None, convert=_ensure_immutable_ids)
INTERNALERROR> Traceback (most recent call last):
INTERNALERROR>   File "/Users/davidchudzicki/.cache/hypothesis-build-runtimes/.tox/py36-full/lib/python3.6/site-packages/_pytest/main.py", line 96, in wrap_session
INTERNALERROR>     config._do_configure()
INTERNALERROR>   File "/Users/davidchudzicki/.cache/hypothesis-build-runtimes/.tox/py36-full/lib/python3.6/site-packages/_pytest/config.py", line 922, in _do_configure
INTERNALERROR>     self.hook.pytest_configure.call_historic(kwargs=dict(config=self))
INTERNALERROR>   File "/Users/davidchudzicki/.cache/hypothesis-build-runtimes/.tox/py36-full/lib/python3.6/site-packages/pluggy/__init__.py", line 628, in call_historic
INTERNALERROR>     res = self._hookexec(self, self._nonwrappers + self._wrappers, kwargs)
INTERNALERROR>   File "/Users/davidchudzicki/.cache/hypothesis-build-runtimes/.tox/py36-full/lib/python3.6/site-packages/pluggy/__init__.py", line 222, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook, methods, kwargs)
INTERNALERROR>   File "/Users/davidchudzicki/.cache/hypothesis-build-runtimes/.tox/py36-full/lib/python3.6/site-packages/pluggy/__init__.py", line 216, in <lambda>
INTERNALERROR>     firstresult=hook.spec_opts.get('firstresult'),
INTERNALERROR>   File "/Users/davidchudzicki/.cache/hypothesis-build-runtimes/.tox/py36-full/lib/python3.6/site-packages/pluggy/callers.py", line 201, in _multicall
INTERNALERROR>     return outcome.get_result()
INTERNALERROR>   File "/Users/davidchudzicki/.cache/hypothesis-build-runtimes/.tox/py36-full/lib/python3.6/site-packages/pluggy/callers.py", line 76, in get_result
INTERNALERROR>     raise ex[1].with_traceback(ex[2])
INTERNALERROR>   File "/Users/davidchudzicki/.cache/hypothesis-build-runtimes/.tox/py36-full/lib/python3.6/site-packages/pluggy/callers.py", line 180, in _multicall
INTERNALERROR>     res = hook_impl.function(*args)
INTERNALERROR>   File "/Users/davidchudzicki/.cache/hypothesis-build-runtimes/.tox/py36-full/lib/python3.6/site-packages/xdist/plugin.py", line 100, in pytest_configure
INTERNALERROR>     from xdist.dsession import DSession
INTERNALERROR>   File "<frozen importlib._bootstrap>", line 961, in _find_and_load
INTERNALERROR>   File "<frozen importlib._bootstrap>", line 950, in _find_and_load_unlocked
INTERNALERROR>   File "<frozen importlib._bootstrap>", line 646, in _load_unlocked
INTERNALERROR>   File "<frozen importlib._bootstrap>", line 616, in _load_backward_compatible
INTERNALERROR>   File "/Users/davidchudzicki/.cache/hypothesis-build-runtimes/.tox/py36-full/lib/python3.6/site-packages/_pytest/assertion/rewrite.py", line 213, in load_module
INTERNALERROR>     py.builtin.exec_(co, mod.__dict__)
INTERNALERROR>   File "/Users/davidchudzicki/.cache/hypothesis-build-runtimes/.tox/py36-full/lib/python3.6/site-packages/xdist/dsession.py", line 5, in <module>
INTERNALERROR>     from xdist.scheduler import (
INTERNALERROR>   File "/Users/davidchudzicki/.cache/hypothesis-build-runtimes/.tox/py36-full/lib/python3.6/site-packages/xdist/scheduler/__init__.py", line 4, in <module>
INTERNALERROR>     from xdist.scheduler.filescope import LoadFileScheduling  # noqa
INTERNALERROR>   File "<frozen importlib._bootstrap>", line 961, in _find_and_load
INTERNALERROR>   File "<frozen importlib._bootstrap>", line 950, in _find_and_load_unlocked
INTERNALERROR>   File "<frozen importlib._bootstrap>", line 646, in _load_unlocked
INTERNALERROR>   File "<frozen importlib._bootstrap>", line 616, in _load_backward_compatible
INTERNALERROR>   File "/Users/davidchudzicki/.cache/hypothesis-build-runtimes/.tox/py36-full/lib/python3.6/site-packages/_pytest/assertion/rewrite.py", line 213, in load_module
INTERNALERROR>     py.builtin.exec_(co, mod.__dict__)
INTERNALERROR>   File "/Users/davidchudzicki/.cache/hypothesis-build-runtimes/.tox/py36-full/lib/python3.6/site-packages/xdist/scheduler/filescope.py", line 1, in <module>
INTERNALERROR>     from . import LoadScopeScheduling
INTERNALERROR> ImportWarning: can't resolve package from __spec__ or __package__, falling back on __name__ and __path__
ERROR: InvocationError: '/bin/bash scripts/basic-test.sh'
_________________________________________________________________________________________________ summary __________________________________________________________________________________________________
ERROR:   py36-full: commands failed
make: *** [check-py36] Error 1
</details>

Probably we could work around that (and any more that come up), but that's I what thought about DeprecationWarning too, and I'm not really up for another adventure like that one right now.

Reply:
Oh, *fiddlesticks*.  Yeah, just stick with the `FutureWarning`-only solution and we'll work the rest out later 😭